lib/spack/spack/cmd/compiler.py

```diff
@@ -84,8 +83,6 @@ def compiler_find(args):
 
     """
     paths = args.add_paths
-    if not paths:
-        paths = get_path('PATH')
```

Comment:
This behavior remains. As I said, OS classes are capable of reading the PATH environment variable. And they do:
https://github.com/LLNL/spack/blob/79c1f7885a7d249eb03b259f1c885a6de90222ce/lib/spack/spack/architecture.py#L261-L262